### PR TITLE
Port Most of the Query Support

### PR DESCRIPTION
**Pull request by masojus**
This all seems to build, but I haven't tried it on MacOS or Linux.

### CODE REVIEW

**Comment by masojus** [comment]
This ports a bunch of Query functionality and tests. We need to exercise it manually very extensively.

**Comment by masojus** [comment]
BTW this brings in a ton of HTTP-related test helpers. I'm hoping we can expand on those.

**Comment by Pothulapati** [comment]
Looks good to me. I also tried some queries using a `netcore2.0 `Console Application. 

Sorry for making you work on this. Mine didn't work because `Queries` wasn't initialised. :\

**Comment by Pothulapati** [comment]
Awesome! Looks like all the Funnel, Multi Analysis are also Ported. What are all left to port now?

File: Keen.NetStandard.Test/QueryTest.cs [hunk 1]
```diff
@@ -0,0 +1,1189 @@
...
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Dynamic;
+using System.Linq;
+using System.Threading.Tasks;
+
+
+namespace Keen.Net.Test
```
Comment (baumatron) [inline Keen.NetStandard.Test/QueryTest.cs]: This should be changed to `Keen.NetStandard.Test`

**Comment by baumatron** [comment]
A couple of tests are failing on MacOS, I can investigate what's going on there.
image

**Comment by baumatron** [comment]
Looks like the serialized JSON only has LF while the expected JSON has CRLF.

**Comment by masojus** [comment]
Did you run this on Linux too or just MacOS? I'd expect it to fail there as well. To start with we should be expecting something like `Environment.NewLine` in those tests, but in some of my older notes I jotted down a TODO to trim all that JSON based on a debug flag.

**Comment by masojus** [comment]
Alright, I rebased and squashed a few things, made a couple fixes, and we've got 117 passing tests on MacOS and Linux, and 316 passing tests across the net35, net45 and netstandard2.0 binaries on Windows. Assuming AppVeyor comes back clean I'll probably move forward with merging this.

**Pull request merged**